modules/ppcp-agentic-commerce/src/Endpoint/ReplaceCartEndpoint.php

```diff
@@ -69,10 +70,28 @@ public function replace_cart( WP_REST_Request $request ): WP_REST_Response {
 			return $this->error( $session );
 		}
 
-		$new_cart = $this->parse_and_validate_cart( $request );
+		$update_data = $this->parse_json_body( $request );
 
-		if ( $new_cart instanceof AgenticError ) {
-			return $this->error( $new_cart );
+		if ( $update_data instanceof AgenticError ) {
+			return $this->error( $update_data );
+		}
+
+		$existing_cart = $session['cart'];
+
+		// Merge update data with existing cart, preserving fields not in the update.
+		// This allows partial updates (e.g., updating items without re-sending payment_method).
+		$new_cart = $existing_cart->with( $update_data );
+
+		$issues = $new_cart->validate();
+		if ( ! empty( $issues ) ) {
+			$issue_details = array_map(
+				function ( $issue ) {
+					return $issue->to_array();
+				},
+				$issues
+			);
+
+			return $this->error( new BadRequestError( 'Cart validation issue', $issue_details ) );
```

Comment:
Same as above: Validation issues must be returned in the `PayPalCartResponse`, not returning an error